modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/upgrade/v1_0_9/PublicationsAdminRoleUpgradeProcess.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.change.tracking.web.internal.upgrade.v1_0_9;
+
+import com.liferay.change.tracking.constants.CTActionKeys;
+import com.liferay.change.tracking.constants.PublicationRoleConstants;
+import com.liferay.change.tracking.model.CTCollection;
+import com.liferay.petra.sql.dsl.DSLQueryFactoryUtil;
+import com.liferay.portal.kernel.model.GroupConstants;
+import com.liferay.portal.kernel.model.ResourceConstants;
+import com.liferay.portal.kernel.model.Role;
+import com.liferay.portal.kernel.model.RoleTable;
+import com.liferay.portal.kernel.service.CompanyLocalService;
+import com.liferay.portal.kernel.service.ResourcePermissionLocalService;
+import com.liferay.portal.kernel.service.RoleLocalService;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+
+import java.util.List;
+
+/**
+ * @author Brooke Dalton
+ */
+public class PublicationsAdminRoleUpgradeProcess extends UpgradeProcess {
+
+	public PublicationsAdminRoleUpgradeProcess(
+		CompanyLocalService companyLocalService,
+		ResourcePermissionLocalService resourcePermissionLocalService,
+		RoleLocalService roleLocalService) {
+
+		_companyLocalService = companyLocalService;
+		_resourcePermissionLocalService = resourcePermissionLocalService;
+		_roleLocalService = roleLocalService;
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		_companyLocalService.forEachCompanyId(
+			companyId -> {
+				List<Role> adminRoles = _roleLocalService.dslQuery(
+					DSLQueryFactoryUtil.select(
+						RoleTable.INSTANCE
+					).from(
+						RoleTable.INSTANCE
+					).where(
+						RoleTable.INSTANCE.name.eq(
+							PublicationRoleConstants.NAME_ADMIN)
+					));
```

Comment:
The name is a unique identifier for the roles, so there should be only one admin role per company. Use `roleLocalService.fetchRole(companyId, roleName)` to get the role.